compiler/rustc_middle/src/middle/region.rs

```diff
@@ -126,6 +127,11 @@ pub enum ScopeData {
     /// Used for variables introduced in an if-let expression.
     IfThen,
 
+    /// Edition 2024: Scope of the condition and then block of an if expression
+    /// Used for variables introduced in an if-let expression,
+    /// whose lifetimes do not cross beyond this scope.
+    IfThen2024,
```

Comment:
Remark [REV3,ED2024]: mentions of Edition 2024.

Reply:
Renamed